src/todo_final.ts

```diff
@@ -0,0 +1,88 @@
+//import type { Config } from "@jest/types";
+//this array will contain all the todo list items
+let todoItems: any[] = [];
+function listAllTodos() {
+	for (var i = 0; i < todoItems.length; i++) {
+		console.log(todoItems[i].id + " " + todoItems[i].text);
+	}
+}
+
+/* This function will create a new todo object based on the
+text that was entered in the text input, and push it into
+the `todoItems` array*/
+function addTodo(text: any, dueAtstr: string) {
+	let todo = {
+		text, //key and value both are text
+		checked: false,
+		id: Date.now(), //The static Date.now() method returns the number of milliseconds elapsed since January 1, 1970
+		createdAt: new Date(), //it will be the current date when the todo is created
+		dueAt: new Date(dueAtstr), //string to date
+	};
+
+	todoItems.push(todo);
+	return todo.id;
+}
+//getter
+function testItems() {
+	return todoItems;
+}
+
+//list completed task
+
+function completeList() {
+	//filter iterate through each item of the array
+
+	let completedItems = todoItems.filter((item) => item.checked === true);
+	for (var i = 0; i < completedItems.length; i++) {
+		console.log(completedItems[i].id + " " + completedItems[i].text);
+	}
+}
+
+//list uncompleted todos
+function uncompletedList() {
+	/// filter is an array method that returns the position of an element
+	// in the array
+
+	const uncompletedItems = todoItems.filter((item) => item.checked === false);
+	for (var i = 0; i < uncompletedItems.length; i++) {
+		console.log(uncompletedItems[i].id + " " + uncompletedItems[i].text);
+	}
+}
+//for user to write the text of the item they want to remove since they dont know abou the id
+function findTodoByText(text: string) {
+	// find the text that matches the text  inside the array
+	const findTodoWithText = todoItems.findIndex(
+		(todoInsideArray) => todoInsideArray.text == text
+	);
+	return todoItems[findTodoWithText]; //return the item not the text itself
+}
+
+//remove todo by text
+
+function removeTodoByText(text: string) {
+	return removeTodo(findTodoByText(text));
+}
+
+function removeTodo(itemsToRemove: any[number]) {
+	// find the index of the todo with the id you are looking for
+	const indexOfTodoToDelete = todoItems.indexOf(
+		(todoInArray: { id: any }) => todoInArray.id === itemsToRemove.id
+	);
```

Comment:
`indexOf` searches for the value directly and doesn't use a testing function like `(todoInArray: { id: any }) => todoInArray.id === itemsToRemove.id`. This function will not work properly. I think you meant to use `findIndex` instead.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf